src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1426,5 +1426,24 @@ private static bool NormalizeTaskTypesCore(CSharpCompilation compilation, ref Ar
             arrayType = arrayType.WithElementType(elementType);
             return true;
         }
+
+        internal static Cci.TypeReferenceWithAttributes GetTypeRefWithAttributes(
+            CSharpCompilation declaringCompilation,
+            TypeSymbol type,
+            Cci.ITypeReference typeRef)
```

Comment:
Or `this Cci.ITypeReference typeRef, ...` with callers using `typeRef.GetTypeRefWithAttributes(DeclaringCompilation, type)`. It does seem strange to pass in `compilation` and `type` though, just to get the attributes from `type` which was why I suggested having the extension method simply return the attributes rather than the `TypeReferenceWithAttributes`.